recipes/boost-ext-ut/all/conanfile.py

```diff
@@ -17,87 +23,83 @@ class UTConan(ConanFile):
 
     @property
     def _minimum_compilers_version(self):
         return {
-            "apple-clang": "11" if tools.Version(self.version) < "1.1.8" else "12",
+            "apple-clang": "11" if Version(self.version) < "1.1.8" else "12",
             "clang": "9",
             "gcc": "9",
-            "msvc": "19",
-            "Visual Studio": "16",
         }
 
+    def layout(self):
+        cmake_layout(self)
+
     def export_sources(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+            copy(self, patch["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
-        if tools.Version(self.version) <= "1.1.8" and self.settings.compiler in ["msvc", "Visual Studio"]:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        if Version(self.version) <= "1.1.8" and is_msvc(self):
             raise ConanInvalidConfiguration("{} version 1.1.8 may not be built with MSVC. "
                                             "Please use at least version 1.1.9 with MSVC.")
-        min_version = self._minimum_compilers_version.get(
-            str(self.settings.compiler))
-        if not min_version:
-            self.output.warn("{} recipe lacks information about the {} "
-                             "compiler support.".format(
-                                 self.name, self.settings.compiler))
-        else:
-            if tools.Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration(
-                    "{} requires C++{} support. "
-                    "The current compiler {} {} does not support it.".format(
-                        self.name, self._minimum_cpp_standard,
-                        self.settings.compiler,
-                        self.settings.compiler.version))
+
+        check_min_vs(self, "192")
+
+        if not is_msvc(self):
+            min_version = self._minimum_compilers_version.get(
+                str(self.settings.compiler))
+            if not min_version:
+                self.output.warn(f"{self.name} recipe lacks information about the {self.settings.compiler} "
+                                 "compiler support.")
+            else:
+                if Version(self.settings.compiler.version) < min_version:
+                    raise ConanInvalidConfiguration(
+                        f"{self.name} requires C++{self._minimum_cpp_standard} support. "
+                        f"The current compiler {self.settings.compiler} {self.settings.compiler.version} does not support it.")
 
     def config_options(self):
-        if tools.Version(self.version) <= "1.1.8":
+        if Version(self.version) <= "1.1.8":
             del self.options.disable_module
 
     def configure(self):
```

Comment:
```suggestion
    def config_options(self):
```

changing default option value based on a settings should be done in `config_options()`